src/reconcile/rec_visibility.c

```diff
@@ -330,8 +330,18 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
                 if (upd->start_ts < r->min_skipped_ts)
                     r->min_skipped_ts = upd->start_ts;
                 continue;
-            } else
+            } else {
+                /*
+                 * For prepared updates written to the date store in salvage, we write the same
+                 * prepared value to the date store. If there is still content for that key left in
+                 * the history store, rollback to stable will bring it back to the data store.
+                 * Otherwise, it removes the key.
+                 */
+                WT_ASSERT(session, F_ISSET(r, WT_REC_EVICT) ||
+                    (F_ISSET(r, WT_REC_VISIBILITY_ERR) &&
+                                     F_ISSET(upd, WT_UPDATE_PREPARE_RESTORED_FROM_DISK)));
```

Comment:
How about a scenario when the `salvage` is called when the prepared update is not written to the data store? Or `salvage` cannot run when there is a prepared transaction? Because I just worry about the assert that you are expecting a prepared update must be restored from disk.

Reply:
Is this possible? Salvage should only run on data from the disk I think. @bvpvamsikrishna

Reply:
If we can call salvage with uncommitted data, the panic check will be triggered all the time.

Reply:
ok, thanks for the details.

Reply:
@bvpvamsikrishna Can you confirm salvage only run on disk data before I merge this?

Reply:
My understanding is salvage only runs on the on-disk data.  @agorrod could you comment?

Reply:
Yes - salvage is based on disk images.